src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -65,9 +65,39 @@ public static void DuplicateKeysFail()
         }
 
         [Fact]
-        public static void DictionaryOfObjectFail()
+        public static void DictionaryOfObject()
```

Comment:
> This should also fix the underlying problem with #37502.

Is there a test we can add to capture such a failure and make sure its fixed? What was the underlying cause of the `ArgumentOutOfRangeException`?

>Serialize an object using `JsonConvert.SerializeObject()` and post it to the service using 

> That being said, I had a look at the JSON being serialized from it by Newtonsoft.Json and it looks like this:

```json
{"Name":"GreeterClient"}
```

@jmezach, can you share the object definition you were using when you hit the issue?

Reply:
The other tests should capture it. I'm not 100% positive as the report didn't have enough data on what was fed into the serializer. We did fail, just not exactly the same way (at least with what I could pull together given the data I had in the report). The failure change is likely due to me adding the initial object support. So much has changed with this path I'm going to rely on validating the scenario more thoroughly once we get ingested by ASP.NET.